client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -114,8 +116,7 @@
                 try {
                     buffer = segmentReader.read(waitTime);
                 } catch (EndOfSegmentException e) {
-                    boolean fetchSuccessors = e.getErrorType().equals(ErrorType.END_OF_SEGMENT_REACHED);
-                    handleEndOfSegment(segmentReader, fetchSuccessors);
+                    handleEndOfSegment(segmentReader);
```

Comment:
The default automaticCheckpoint interval is 120 seconds (`ReaderGroupConfig#automaticCheckpointIntervalMillis`), this would imply that reader would not read successor segments until the checkpoint has completed. (i.e. the reader can only acquire newer unassigned segments, that belong to a different keyspace until the checkpoint completes) . This would cause a slow down of the reader during a scale up of a stream. Is my understanding correct here?

Reply:
yes. Good point. I will lower the default interval.

Reply:
There is a bound on the number of concurrent checkpoints, so checkpointing that frequently will almost certainly throw an exception. In general a reader is not going to notice a checkpoint any sooner than the refresh interval which is by default 3 seconds. This of course can also be adjusted, but it seems unlikely a checkpoint would be able to reach all readers and get the results in under 100ms. However even attempting to achieve this is probably not a good idea as it would impose significant overhead, which would impact throughput and waste resources.